packages/ckeditor5-image/src/imageresize/imageresizehandles.js

```diff
@@ -74,13 +78,53 @@ export default class ImageResizeHandles extends Plugin {
 
 			resizer.on( 'updateSize', () => {
 				if ( !widget.hasClass( 'image_resized' ) ) {
-					editor.editing.view.change( writer => {
+					editingView.change( writer => {
 						writer.addClass( 'image_resized', widget );
 					} );
 				}
 			} );
 
+			this._hideResizerUntilImageIsLoaded( resizer, widget );
+
 			resizer.bind( 'isEnabled' ).to( this );
 		}, { priority: 'low' } );
 	}
+
+	/**
+	 * Function to add listeners that will hide the resizer frame until the image is loaded,
+	 * to avoid [resize frame flashing when the image is not yet loaded](https://github.com/ckeditor/ckeditor5/issues/8088).
+	 *
+	 * @private
+	 * @param {module:widget/widgetresize/resizer~Resizer} resizer
+	 * @param {module:engine/view/containerelement~ContainerElement} widget
+	 */
+	_hideResizerUntilImageIsLoaded( resizer, widget ) {
+		// Given that #7548 is fixed, this logic might no longer be needed.
+		const editor = this.editor;
+		const editingView = editor.editing.view;
+
+		editingView.change( writer => {
```

Comment:
So we first attach a resizer to the image and then hide it. Why we don't attach a hidden resizer? The order seems to be wrong here.

Also, maybe the image could change the class when it's loading and loaded by itself? That may be helpful in other features. So, what I mean is that this code could be part of the core image feature or a new plugin that we could easily reuse.

Reply:
BTW, you could check where this observer is already used and check whether rewriting this as a plugin could not help.

Reply:
Actually that was a good point, I changed the approach further and instead of playing with HTML classes I'm delaying resizer creation until the image is loaded.